src/Alert.js

```diff
@@ -0,0 +1,96 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
+
+const propTypes = {
+  className: PropTypes.any,
+  color: PropTypes.string,
+  onDismiss: PropTypes.func,
+  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string])
+};
+
+const defaultProps = {
+  color: 'warning',
+  tag: 'div'
+};
+
+export const Alert = (props) => {
+  const {
+    className,
+    tag: Tag,
+    color,
+    onDismiss,
+    ...attributes
+  } = props;
+
+  const classes = classNames(
+    className,
+    'alert',
+    `alert-${color}`,
+    { 'alert-dismissible': onDismiss }
+  );
+
+  return (
+    <Tag {...attributes} className={classes} role="alert">
+      { onDismiss ?
+        <button type="button" className="close" aria-label="Close" onClick={onDismiss}>
+          <span aria-hidden="true">&times;</span>
+        </button>
+        : null }
+      {props.children}
+    </Tag>
+  );
+};
+
+Alert.propTypes = propTypes;
+Alert.defaultProps = defaultProps;
+
+const wrapperPropTypes = {
+  dismissible: PropTypes.bool
+}
+
+const wrapperDefaultPropTypes = {
+  dismissible: false
+}
+
+class AlertWrapper extends React.Component {
```

Comment:
it might just be one component if you move the `ReactCSSTransitionGroup` up to `Alert`. Also, you could pass `tag` through as the `component` prop of [ReactCSSTransitionGroup](https://facebook.github.io/react/docs/animation.html#rendering-a-different-component).